spec/account/account-manager.spec.js

```diff
@@ -0,0 +1,154 @@
+        otDateMock = require("../mock/ot-date-mock");
+        promiseMock = require("../mock/promise-mock");
+        randomMock = require("../mock/random-mock");
+        class AccountServiceFake {
+            constructor() {
+                this.initiate = jasmine.createSpy("initiate");
+                this.initiate.andCallFake((accountId, accountInfo, options) => {
+                    return promiseMock.resolve({
+                        accountId: accountInfo.accountId,
+                        email: accountInfo.email,
+                        mfa: accountInfo.mfa,
+                        salt: accountInfo.salt
+                    });
+                });
+                this.complete = jasmine.createSpy("complete");
+                this.complete.andCallFake(() => {
+                    return promiseMock.resolve(true);
+                });
+                this.get = jasmine.createSpy("get");
+                this.get.andCallFake(() => {
+                    return promiseMock.resolve({
+                        mfa: "339r93939303093"
+                    });
+                });
+            }
+        }
+        class HotpFake {
+            constructor() {
+                this.generateSecret = jasmine.createSpy("generateSecret");
+                this.generateSecret.andCallFake(() => {
+                    return promiseMock.resolve("thisisasecrectcodefrommfa");
+                });
+                // This is a sync call
+                this.verifyToken = jasmine.createSpy();
+                this.verifyToken.andCallFake((key, token, opts) => {
+                    if (token === "987654") {
+                        return false;
+                    }
+
+                    return true;
+                });
+            }
+        }
+        config = {
+            account: {
+                completeLifetime: {
+                    months: 6
+                },
+                initiateLifetime: {
+                    hours: 1
+                }
+            },
+            hotp: {
+                previous: {
+                    afterDrift: 1,
+                    beforeDrift: 1,
+                    drift: 2
+                }
+            }
+        };
+        accountManager = new AccountManager(new AccountServiceFake, config, new HotpFake, otDateMock, randomMock, promiseMock);
+    });
+    describe(".initiate()", () => {
+        it("gets an object back from service", (done) => {
+            accountManager.initiate({
+                email: "some.one@example.net"
+            }).then((result) => {
+                expect(result).toEqual({
+                    accountId: jasmine.any(Buffer),
+                    email: "some.one@example.net",
+                    mfa: "thisisasecrectcodefrommfa",
+                    salt: jasmine.any(Buffer)
+                });
+                expect(result.accountId.length).toBe(24);
+                expect(result.salt.length).toBe(128);
+
+            }).then(done, done);
```

Comment:
I don't understand how this is verifying that it called the service.  The manager's responsibility is only to apply business logic.  It only verifies that the tokens are right and sends an expires value to the service.  Nothing in here appears to be testing that functionality.  So you have the test coverage but appear to have missed the point of the function.